byway/src/main/webapp/destinations.js

```diff
@@ -218,21 +237,69 @@ function getCurrentAddress(){
 * add event listener for submit button
 */
 window.onload = function(){
+  
   let nextButton = document.getElementById('next-button');
   nextButton.addEventListener('click', () => {
-      nextButton.href = buildUrlWithParams("/interests.html", {tripKey});
+      nextButton.href = configureTripKeyForPath(tripKey, '/interests.html');
   });
   document.getElementById('user-input-form').addEventListener('submit', (event) => {
     event.preventDefault();
-    const formData = new  FormData(document.getElementById("user-input-form"));
-    fetch(buildUrlWithParams("/api/destinations", {tripKey}), {method: 'POST', body:formData}).then((response)=>
-        response.json()).then(locationData => {
-        updateLocations(locationData);
-        updateStartDestination(locationData);
-      });
-    });
+    savePlaceIds();
+  });
 }
 
+/** 
+* gets User Input from search box and sends the Place Id of user input to the server
+* elementName indicates which search box to get from, param indicates which param to save as
+* @param {String} elementName
+*/
+function savePlaceIds(){
+  let formData = new FormData();
+
+  const destRequest = {
+    query: String(document.getElementById("destinations-search-box").value),
+    fields: ["place_id"]
+  };
+  const startRequest = {
+    query: String(document.getElementById("start-search-box").value),
+    fields: ["place_id"]
+  };
+
+  let promise = new Promise((resolve,reject) => {
+    placesService.findPlaceFromQuery(destRequest, (results, status) => {
+    console.log(status);
+      if (status === google.maps.places.PlacesServiceStatus.OK) {
+        formData.append("destinations-search-box", results[0].place_id);
+        resolve(results);
+      }
+      else{
+        alert("Status: " + status);
+        reject(new Error("Could not find place:" + status));
+      }
+    });
+  }).catch(error => {
+    alert("Error: cannot process this request due to " + error);
+  });
```

Comment:
There's no point in rejecting if you're immediately going to catch your own error. Consider what this would look like if it was not using promises:

```
try {
  if (status == "OK") {
    return result; // resolve = return
  } else {
    alert("Status: " + status);
    throw new Error("..."); // reject = throw
  }
} catch (error) {
    alert(error);
}
```

Look at what happens in the error case:

1) `alert("Status: " + status);`
2) `alert(new Error("..."));` throw wrapped directly in catch + alert is basically equivalent to just sticking the error into the alert.
3) `return undefined;` Since there's no explicit return in the `catch` case, it is the same as returning `undefined`.

Reply:
removed the reject